File: frontend/src/models/content.ts
export class Content {
    isText: boolean;
    isCodeBlock: boolean;
    isFile: boolean;
    text?: string;
    fileName?: string;
    formFile?: File | null;
    fileUrl?: string;
    fileExtension?: string;
    codeLanguage?: string;

    constructor(
        isText: boolean,
        isCodeBlock: boolean,
        isFile: boolean,
        text?: string,
        codeLanguage?: string,
        formFile?: File | null
    ) {
        this.isText = isText;
        this.isCodeBlock = isCodeBlock;
        this.isFile = isFile;
        this.text = text;
        this.codeLanguage = codeLanguage;
        this.formFile = formFile;
    }
}

File: LearnLink.Core/Entities/ContentEntities/Content.cs
namespace LearnLink.Core.Entities
{
    public class Content
    {
        public bool IsText { get; set; }
        public bool IsCodeBlock { get; set; }
        public bool IsFile { get; set; }

        public string? Text { get; set; }
        public string? FileName { get; set; }
        public string? FileExtension { get; set; }
        public string? CodeLanguage { get; set; }
    }
}
